src/components/Web3ReactManager/index.tsx

```diff
@@ -48,6 +53,20 @@ export default function Web3ReactManager({ children }: { children: JSX.Element }
     }
   }, [])
 
+  useEffect(() => {
+    // start listeners for affiliate link generation on account change
+    startWalletListeners()
+  }, [active])
+
+  const startWalletListeners = async () => {
+    const provider: any = await detectEthereumProvider()
+    if (!provider && active) return
+    provider.on('accountsChanged', function (accounts: Array<string>) {
+      console.log('ACCOUNT CHANGED', accounts)
+      generateAffiliateLink((link: string) => dispatch(updateAffiliateLink({ affiliateLink: link })))
+    })
+  }
+
```

Comment:
I'm not sure why this handler is needed to detect an account change. Why not just use the `account` property of `useWalletInfo` in a useEffect?